Classes/EventListener/ProcessRequestTokenListener.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Causal\Oidc\EventListener;
+
+use TYPO3\CMS\Core\Authentication\Event\BeforeRequestTokenProcessedEvent;
+use TYPO3\CMS\Core\Security\RequestToken;
+
+final class ProcessRequestTokenListener
+{
+    public function __invoke(BeforeRequestTokenProcessedEvent $event): void
+    {
+        $user = $event->getUser();
+        $requestToken = $event->getRequestToken();
+        if ($requestToken instanceof RequestToken) {
+            // fine, there is a valid request-token
+            return;
+        }
+        // @TODO how to improve security? Maybe get rid of redirects?
```

Comment:
What do you mean by this todo?

Reply:
This TODO comes from the original PR that I rebased.

According to the [documentation](https://docs.typo3.org/m/typo3/reference-coreapi/main/en-us/ApiOverview/Authentication/CSRFlikeRequestTokenHandling.html), the workflow is as follows:
Create token -> Render form -> Submit token via POST -> Evaluate token

However, the [feature log for BeforeRequestTokenProcessedEvent](https://docs.typo3.org/c/typo3/cms-core/main/en-us/Changelog/12.0/Feature-97305-IntroduceCSRF-likeRequest-tokenHandling.html#intercept-adjust-request-token) also states:
"Scenarios that are not using a login callback and do not have the possibility to submit a request token can use :php:\TYPO3\CMS\Core\Authentication\Event\BeforeRequestTokenProcessedEvent to generate the token individually."

You might be in a better position to judge whether there is a better way to create the request token in advance. Perhaps it should be stored as a cookie before redirecting to the OIDC provider?

Reply:
Let's remove the todo. As far as I debugged this, I see no way to create this somewhere earlier.